### PR TITLE
Check for StorageFillComponent before initializing item mapped layers

### LINKED ISSUE #10622: Sprites such as toolbelts or crayon boxes aren't showing properly
## Description
Something broke this recently, as this used to work 

Sometimes a full box of crayons will show as empty like this, until you open it
closed:
image
after open:
image

This also happens to toolbelts, it will show the sprite for a toolbelt without any tools in it until you open it.

### PR DESCRIPTION
**Pull request by JustinTrotter**
## About the PR 
Fixes #10622 
Just needed to check if it's an entity with a StorageFillComponent to determine if it should have the layers on initialization.

**Screenshots**

**Changelog**

:cl:
- fix: Filled containers like toolbelts and crayon boxes properly show their contents on spawn

### CODE REVIEW

File: Content.Client/Storage/Systems/ItemMapperSystem.cs [hunk 1]
```diff
@@ -41,13 +43,14 @@ private void InitLayers(ItemMapperComponent component, SpriteComponent spriteCom
...
 
+        TryComp<StorageFillComponent>(component.Owner, out var storageFill);
         component.SpriteLayers.AddRange(wrapper.QueuedEntities);
 
         foreach (var sprite in component.SpriteLayers)
         {
             spriteComponent.LayerMapReserveBlank(sprite);
             spriteComponent.LayerSetSprite(sprite, new SpriteSpecifier.Rsi(component.RSIPath!, sprite));
-            spriteComponent.LayerSetVisible(sprite, false);
+            spriteComponent.LayerSetVisible(sprite, storageFill != null);
```
Comment (mirrorcult) [inline Content.Client/Storage/Systems/ItemMapperSystem.cs:53]: what does this even fix? this feels more like an init order bug than anything else (storagefill not spawning before itemmapper does). tying this behavior specifically to storagefill feels like it is going to introduce even more bugs

**Review by mirrorcult — CHANGES_REQUESTED** [review]

File: Content.Client/Storage/Systems/ItemMapperSystem.cs (hunk 1 above)
Comment (JustinTrotter) [reply Content.Client/Storage/Systems/ItemMapperSystem.cs:52]: I'm back after a two week absence. While playing last night I noticed other issues with placing items into the belt weren't updating the worn belt's display. I'll play with it some more to see if there's alternatives to fixing this or if changing it here is necessary.

Comment (JustinTrotter) [reply Content.Client/Storage/Systems/ItemMapperSystem.cs:52]: Yeah, it's not addressing this other issue I'm noticing. I'll take a different approach and see if I can fix both issues.

**Comment by wrexbe** [comment]
Can't reproduce the bug. Might have been fixed by something else.

**Comment by mirrorcult** [comment]
almost certainly was caused & fixed by the appearance changes recently

**Pull request closed without merging**